### ISSUE TITLE
Cruelty stops working when you change into certain weapons and relog

### ISSUE THREAD

**Issue by magey** [magey/tbc-warrior#65]
Cruelty stops giving its effect when you change into certain weapons and log off and back on again:

1. Spec 5/5 Cruelty, you gain 5% crit
2. Log off and back on, still retain the crit bonus
3. Change into one of the weapons which trigger the bug, still retain the crit bonus
4. Log off and back on with the new weapon - the crit bonus is gone
5. Change weapons again and log off and back on - you regain the crit bonus

You can keep doing steps 3-5 over and over and keep losing/regaining the crit bonus. Example video using Mag'hari Battleaxe to trigger the bug:

https://user-images.githubusercontent.com/44882404/118417649-c380bb80-b6bd-11eb-96f4-876bf3aa70ce.mp4

**Comment by Scyllaqt** [comment]
I've been able to reproduce this bug consistently with my 70 human warrior by equipping white weapons (no stats).
https://streamable.com/dfb35z

What I've noticed so far:
1. Swapping to a white weapon and relogging causes you to lose 5% crit
2. If you swap to a weapon with stats and relog, you regain the lost crit
3. Swapping to a different white weapon and relogging is NOT giving me back the crit

VERY STRANGELY, this bug stopped occurring on my warrior out of nowhere despite being reproducible on demand for an hour.  Unlearning my talents and respeccing back into cruelty didn't cause the bug to occur again either.  Making a completely new warrior to spec into cruelty on DID make the issue reappear however, and my above findings are still holding true on it. No idea what's going on with that.
